inginious/frontend/plugins/statistics/pages/user_statistics.html

```diff
@@ -21,12 +21,24 @@ <h3>Bar Submissions Per Tasks</h3>
                   Toggle normalize
           </button>
         </div>
-        <a href='#' onclick='UserStatistics.barSubmissionsPerTasks.downloadCsvAsync();'>Download CSV</a>
+        <a href='#' class="download_csv" onclick='UserStatistics.barSubmissionsPerTasks.downloadCsvAsync();'>Download CSV</a>
     </div>
 </div>
 
 <script>
     function getCourseId() {
        return "$:course_id";
     }
+
+    // Checking if csv download function exist in the code
+    var download_csv_buttons = document.getElementsByClassName('download_csv');
+    for(let it = 0; it < download_csv_buttons.length; it++){
+        element = download_csv_buttons[it];
+        fun_name = element.getAttribute('onclick');
+
+        // windown['function']. To use check function from a string.
+        if( window[fun_name] !== "function" ) {
+            element.style.visibility = "hidden";
```

Comment:
The same here, I do not understand very well when this should work as this was hidden to me testing this in my statistics and course statistics. Maybe we should show the option when there's not data in the statistics. But please, comment the use cases for this.

Reply:
This piece of code checks if there exists a function with the name fun_name. In this case is the function that is suppose to run after the click of the link.

So at the end the function checks that this link can run a js function. And shows it or hides it, depending of the case